### PR TITLE
LPS-88715 Integration tests for media object API

### PR DESCRIPTION
**Pull request by ruben-pulido**

### CODE REVIEW

File: modules/apps/headless-apio/media-object/media-object-apio-test/build.gradle [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+dependencies {
+	compileOnly group: "com.fasterxml.jackson.core", name: "jackson-databind", version: "2.6.7.1"
+	compileOnly group: "org.apache.cxf", name: "cxf-rt-frontend-jaxrs", version: "3.2.4"
+	compileOnly group: "org.assertj", name: "assertj-core", version: "3.10.0"
+	compileOnly project(":apps:headless-apio:portal:portal-apio-test-util")
+
+	testIntegrationCompile group: "com.jayway.jsonpath", name: "json-path", version: "2.2.0"
+	testIntegrationCompile group: "com.liferay.portal", name: "com.liferay.portal.kernel", version: "default"
+	testIntegrationCompile group: "org.apache.httpcomponents", name: "httpasyncclient", version: "4.1.3"
+	testIntegrationCompile group: "org.apache.httpcomponents", name: "httpclient", version: "4.5.3"
```
Comment (cgoncas) [inline modules/apps/headless-apio/media-object/media-object-apio-test/build.gradle]: Do we still need the httpcomponents?

File: modules/apps/headless-apio/media-object/media-object-apio-test/src/testIntegration/java/com/liferay/media/object/apio/internal/architect/resource/test/DefaultMediaObjectNestedCollectionResourceTest.java [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+					"application/octet-stream"),
+				"My media object testGetPageItems",
+				"My media object description", null, null));
+
+		PageItems<FileEntry> pageItems = getPageItems(
+			PaginationRequest.of(10, 1), _group.getGroupId());
+
+		Assert.assertEquals(1, pageItems.getTotalCount());
+
+		Collection<FileEntry> dlMediaObjects = pageItems.getItems();
```
Comment (cgoncas) [inline modules/apps/headless-apio/media-object/media-object-apio-test/src/testIntegration/java/com/liferay/media/object/apio/internal/architect/resource/test/DefaultMediaObjectNestedCollectionResourceTest.java]: This should be a list, right?

File: modules/apps/headless-apio/media-object/media-object-apio-test/src/testIntegration/java/com/liferay/media/object/apio/internal/architect/resource/test/BaseMediaObjectNestedCollectionResourceTest.java [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+public class BaseMediaObjectNestedCollectionResourceTest {
+
+	protected FileEntry addFileEntry(
+			long groupId, ServiceContext serviceContext)
+		throws Exception {
+
+		return DLAppServiceUtil.addFileEntry(
+			groupId, DLFolderConstants.DEFAULT_PARENT_FOLDER_ID, null, null,
+			RandomTestUtil.randomString(10), RandomTestUtil.randomString(10),
+			StringPool.BLANK, new ByteArrayInputStream(_CONTENT.getBytes()), 0L,
```
Comment (cgoncas) [inline modules/apps/headless-apio/media-object/media-object-apio-test/src/testIntegration/java/com/liferay/media/object/apio/internal/architect/resource/test/BaseMediaObjectNestedCollectionResourceTest.java]: Use the content as a parametter, do not hide the file you are trying to create.

If you don't care, use a Random

File: modules/apps/headless-apio/media-object/media-object-apio-test/src/testIntegration/java/com/liferay/media/object/apio/internal/architect/resource/test/BaseMediaObjectNestedCollectionResourceTest.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+	protected FileEntry addMediaObject(long groupId, MediaObject mediaObject)
+		throws Exception {
+
+		NestedCollectionResource nestedCollectionResource =
+			_getNestedCollectionResource();
+
+		Class<?> clazz = nestedCollectionResource.getClass();
+
+		Method method = clazz.getDeclaredMethod(
+			"_addFileEntry", long.class, MediaObject.class);
```
Comment (cgoncas) [inline modules/apps/headless-apio/media-object/media-object-apio-test/src/testIntegration/java/com/liferay/media/object/apio/internal/architect/resource/test/BaseMediaObjectNestedCollectionResourceTest.java]: The method it is called addMediaObject, but it calling the _addFileEntry method

File: modules/apps/headless-apio/media-object/media-object-apio-test/src/testIntegration/java/com/liferay/media/object/apio/internal/architect/resource/test/DefaultMediaObjectNestedCollectionResourceTest.java [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+		_group = GroupTestUtil.addGroup();
+	}
+
+	@Test
+	public void testAddMediaObject() throws Exception {
+		FileEntry fileEntry = addMediaObject(
+			_group.getGroupId(),
+			new MediaObjectImpl(
+				new BinaryFile(
+					new ByteArrayInputStream(_CONTENT.getBytes(UTF_8)), 0L,
```
Comment (cgoncas) [inline modules/apps/headless-apio/media-object/media-object-apio-test/src/testIntegration/java/com/liferay/media/object/apio/internal/architect/resource/test/DefaultMediaObjectNestedCollectionResourceTest.java]: You are not asserting the content, use random or assert it

File: modules/apps/headless-apio/media-object/media-object-apio-test/src/testIntegration/java/com/liferay/media/object/apio/internal/architect/resource/test/DefaultMediaObjectNestedCollectionResourceTest.java [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+					"application/octet-stream"),
+				"My media object testGetFileEntryPreviewURL", null, null,
+				null));
+
+		String fileEntryPreview = getFileEntryPreviewURL(fileEntry);
+
+		Assert.assertNotNull(fileEntryPreview);
```
Comment (cgoncas) [inline modules/apps/headless-apio/media-object/media-object-apio-test/src/testIntegration/java/com/liferay/media/object/apio/internal/architect/resource/test/DefaultMediaObjectNestedCollectionResourceTest.java]: Just notNull? Couldn't we test anything about the entryPreview?

File: modules/apps/headless-apio/media-object/media-object-apio-test/src/testIntegration/java/com/liferay/media/object/apio/internal/architect/resource/test/DefaultMediaObjectNestedCollectionResourceTest.java [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+
+	@Test
+	public void testGetMediaObjectAssetTags() throws Exception {
+		List<String> keywords = Arrays.asList("keyword1", "keyword2");
+
+		FileEntry fileEntry = addMediaObject(
+			_group.getGroupId(),
+			new MediaObjectImpl(
+				new BinaryFile(
+					new ByteArrayInputStream(_CONTENT.getBytes(UTF_8)), 0L,
```
Comment (cgoncas) [inline modules/apps/headless-apio/media-object/media-object-apio-test/src/testIntegration/java/com/liferay/media/object/apio/internal/architect/resource/test/DefaultMediaObjectNestedCollectionResourceTest.java]: If we don't care about the content, use random

File: modules/apps/headless-apio/media-object/media-object-apio-test/src/testIntegration/java/com/liferay/media/object/apio/internal/architect/resource/test/DefaultMediaObjectNestedCollectionResourceTest.java [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+			mediaObjectAssetTags.containsAll(keywords));
+	}
+
+	@Test
+	public void testGetPageItems() throws Exception {
+		addMediaObject(
+			_group.getGroupId(),
+			new MediaObjectImpl(
+				new BinaryFile(
+					new ByteArrayInputStream(_CONTENT.getBytes(UTF_8)), 0L,
```
Comment (cgoncas) [inline modules/apps/headless-apio/media-object/media-object-apio-test/src/testIntegration/java/com/liferay/media/object/apio/internal/architect/resource/test/DefaultMediaObjectNestedCollectionResourceTest.java]: If we don't test the _CONTENT, use random

File: modules/apps/headless-apio/media-object/media-object-apio-test/src/testIntegration/java/com/liferay/media/object/apio/internal/architect/resource/test/DefaultMediaObjectNestedCollectionResourceTest.java [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+
+		private MediaObjectImpl(
+			BinaryFile binaryFile, String title, String description,
+			List<String> keywords, List<Long> categories) {
+
+			_binaryFile = binaryFile;
+			_title = title;
+			_description = description;
+			_keywords = keywords;
+			_categories = categories;
```
Comment (cgoncas) [inline modules/apps/headless-apio/media-object/media-object-apio-test/src/testIntegration/java/com/liferay/media/object/apio/internal/architect/resource/test/DefaultMediaObjectNestedCollectionResourceTest.java]: The list should always be copied, never set with a reference.

File: modules/apps/headless-apio/media-object/media-object-apio-test/src/testIntegration/java/com/liferay/media/object/apio/internal/architect/resource/test/BaseMediaObjectNestedCollectionResourceTest.java [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * @author Ruben Pulido
+ */
+public class BaseMediaObjectNestedCollectionResourceTest {
+
+	protected FileEntry addFileEntry(
```
Comment (cgoncas) [inline modules/apps/headless-apio/media-object/media-object-apio-test/src/testIntegration/java/com/liferay/media/object/apio/internal/architect/resource/test/BaseMediaObjectNestedCollectionResourceTest.java]: This method is only called in ResourcePermissionMediaObjectNestedCollectionResourceTest, please move to the right class

File: modules/apps/headless-apio/media-object/media-object-apio-test/src/testIntegration/java/com/liferay/media/object/apio/internal/architect/resource/test/DefaultMediaObjectNestedCollectionResourceTest.java (hunk 6 above)
Comment (ruben-pulido) [reply modules/apps/headless-apio/media-object/media-object-apio-test/src/testIntegration/java/com/liferay/media/object/apio/internal/architect/resource/test/DefaultMediaObjectNestedCollectionResourceTest.java]: After discussion, we'll test the content returned by contentUrl in client tests

**Comment by ruben-pulido** [comment]
Thanks for the feedback @cgoncas! Feedback incorporated [here](https://github.com/cgoncas/liferay-portal/pull/343)

**Comment by ruben-pulido** [comment]
Sent to Brian [here](https://github.com/brianchandotcom/liferay-portal/pull/66349), after incorporating feedback and fixing SF issues.

**Pull request closed without merging**